File: src/main/java/edu/uchicago/gerber/_08final/mvc/model/NewShipFloater.java
package edu.uchicago.gerber._08final.mvc.model;

import edu.uchicago.gerber._08final.mvc.controller.Game;

import java.awt.*;
import java.util.ArrayList;
import java.util.List;

public class NewShipFloater extends Sprite {


	public NewShipFloater() {

		super();
		setTeam(Team.FLOATER);

		setExpiry(250);
		setRadius(50);
		setColor(Color.BLUE);

		//set random DeltaX
		setDeltaX(somePosNegValue(10));

		//set rnadom DeltaY
		setDeltaY(somePosNegValue(10));
		
		//set random spin
		setSpin(somePosNegValue(10));

		//random orientation 
		setOrientation(Game.R.nextInt(360));

		//always set cartesian points last
		List<Point> pntCs = new ArrayList<>();
		pntCs.add(new Point(5, 5));
		pntCs.add(new Point(4,0));
		pntCs.add(new Point(5, -5));
		pntCs.add(new Point(0,-4));
		pntCs.add(new Point(-5, -5));
		pntCs.add(new Point(-4,0));
		pntCs.add(new Point(-5, 5));
		pntCs.add(new Point(0,4));

		setCartesians(pntCs);
	}

	@Override
	public void move() {
		super.move();
		//a newShipFloater spins
		setOrientation(getOrientation() + getSpin());
		//and it also expires
		expire();

	}


}

File: src/main/java/edu/uchicago/gerber/_08final/mvc/model/NewShipFloater.kt
package edu.uchicago.gerber._08final.mvc.model

import edu.uchicago.gerber._08final.mvc.controller.Utils
import edu.uchicago.gerber._08final.mvc.model.Movable.Team
import java.awt.Color
import java.awt.Graphics
import java.awt.Point
import java.util.*

class NewShipFloater : Sprite() {
    init {
        team = Team.FLOATER
        expiry = 251
        radius = 50
        color = Color.BLUE

        //set random DeltaX
        deltaX = somePosNegValue(10).toDouble()

        //set rnadom DeltaY
        deltaY = somePosNegValue(10).toDouble()

        //set random spin
        spin = somePosNegValue(10)

        //cartesian points which define the shape of the polygon
        val listPoint = ArrayList<Point>()
        listPoint.add(Point(5, 5))
        listPoint.add(Point(4, 0))
        listPoint.add(Point(5, -5))
        listPoint.add(Point(0, -4))
        listPoint.add(Point(-5, -5))
        listPoint.add(Point(-4, 0))
        listPoint.add(Point(-5, 5))
        listPoint.add(Point(0, 4))
        cartesians = Utils.pointsListToArray(listPoint)
    }

    override fun draw(g: Graphics) {
        renderVector(g)
    }


}
